test/test_sparse_csr.py

```diff
@@ -614,156 +614,163 @@ def test_validate(self, layout, device, dtype):
                     layout, device, dtype, index_dtype, enable_batched=True, enable_hybrid=True):
                 torch._validate_sparse_compressed_tensor_args(compressed_indices, plain_indices, values, size, layout)
 
-    def _generate_invalid_input(self, layout):
+    def _generate_invalid_input(self, layout, device):
+        from functools import partial
 
-        shape, values = self._generate_small_inputs_utils(layout)
+        shape, values = self._generate_small_inputs_utils(layout, device=device)
+
+        tensor = partial(torch.tensor, device=device)
+        values = partial(values, device=device)
```

Comment:
Add missing tests:
```suggestion
        values = partial(values, device=device)

        yield ('shape_mismatch_compressed_indices',
               tensor([0, 2, 4]),
               tensor([0, 1, 0, 2]),
               values([1, 2, 3, 4]),
               shape((3, 3)),
               'compressed_indices have wrong shape: ....')
        yield ('shape_mismatch_plain_indices',
               tensor([0, 2, 4]),
               tensor([0, 1, 0, 2]),
               values([1, 2, 3, 4, 5]),
               shape((2, 3)),
               'plain_indices have wrong shape: ....')
```

Reply:
These are checked in the enclosing function.